feature-generation/feature-generation/script.whizzml

```diff
@@ -0,0 +1,216 @@
+    (cond (= match "antecedent")
+            (join " and " lhs)
+          (= match "consequent")
+            (join " and " rhs)
+            (str (join " and " lhs) " then " (join " and " rhs)))))
+
+(define (flatline-rule association rule match)
+  (let (fields (resource-fields association)
+        items (association ["associations" "items"] [])
+        get-item-fn (lambda (i)
+                      (flatline-item (items i) (fields ((items i) "field_id"))))
+        lhs (map get-item-fn (rule "lhs"))
+        rhs (map get-item-fn (rule "rhs"))
+        lhs (if (> (count lhs) 1)
+                (str "(and " (join " " lhs) ")")
+                (join " " lhs))
+        rhs (if (> (count rhs) 1)
+                (str "(and " (join " " rhs) ")")
+                (join " " rhs)))
+    (cond (= match "antecedent")
+            lhs
+          (= match "consequent")
+            rhs
+            (str "(and " lhs " " rhs ")"))))
+
+(define (batch-association-sets dataset-id association-id dataset-name
+         match . rule-ids)
+  (let (association (fetch association-id)
+        dataset-name (if (empty? dataset-name)
+                         (str ((fetch dataset-id) "name") " with "
+                              ((fetch association-id) "name"))
+                         dataset-name)
+        rule-ids (if (empty? (head rule-ids []))
+                   (map (lambda (r) (r "id"))
+                        (association ["associations" "rules"]))
+                   (head rule-ids))
+        rules (filter (lambda (r) (member? (r "id") rule-ids))
+                      (association ["associations" "rules"]))
+        flatline-rules (map (lambda (r) (flatline-rule association r match))
+                            rules)
+        new-fields (iterate (acc [] flatline-rule flatline-rules rule rules)
+                     (append acc {"field" flatline-rule
+                                  "name" (describe-rule association
+                                                        (rule "id")
+                                                        match)
+                                  "label" (str (association "name")
+                                               " - rule "
+                                               (rule "id"))})))
+    (when (empty? new-fields)
+          (raise (str "No rules were found for the list of IDs: "
+                      (join ", " rule-ids))))
+    (create-dataset {"origin_dataset" dataset-id
+                     "name" dataset-name
+                     "new_fields" new-fields})))
+
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; own code for the rest of unsupervised models ;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(define (get-exec-model-id exec model-type)
+  (let (execution (fetch (wait exec))
+       outputs (filter (lambda (r) (= (r 2) model-type))
+                  (get-in execution ["execution" "outputs"])))
+    (when (not (empty? outputs))
+       ((head outputs) 1))))
```

Comment:
could that fail if someone passes a different execution by mistake? maybe checking that the outputs have the expected length and type and raising the error otherwise